sleap/gui/app.py

```diff
@@ -1464,6 +1436,65 @@ def _update_track_menu(self):
             "New Track", self.commands.addTrack, Qt.CTRL + Qt.Key_0
         )
 
+    def _update_sessions_menu(self, frame_idx):
+            """Update the instance groups menu based on the selected frame index in the session."""
+            session = self.state.get("session")
+            if session is None:
+                return 
+
+
+            frame_idx = self.state["frame_idx"]
+            frame_group = session.frame_groups.get(frame_idx, None)
+            if frame_group is not None:
+                for inst_group_ind, instance_group in enumerate(frame_group.instance_groups):
+                    if inst_group_ind < 9:
+                        key_command = Qt.SHIFT + Qt.Key_0 + inst_group_ind + 1
+                    else:
+                        key_command = ""
+        
+                    self.inst_groups_menu.addAction(
+                        instance_group.name,
+                        lambda x=instance_group: self.commands.setInstanceGroup(x),
+                        shortcut=key_command
+                    )
+    
+            self.inst_groups_menu.addAction(
+                "New Instance Group",
+                self.commands.addInstanceGroup,
+                shortcut=Qt.SHIFT + Qt.Key_0
+            )
+
+    def setup_menus(self):
+        self.inst_groups_delete_menu = QMenu("Delete Instance Group", self.sessionsMenu)
+        self.sessionsMenu.addMenu(self.inst_groups_delete_menu)
+
+        self.inst_groups_delete_menu.aboutToShow.connect(self.update_delete_instance_group_menu)
+
+    def update_delete_instance_group_menu(self):
+        self.inst_groups_delete_menu.clear()
+        session = self.state["session"]
+        frame_idx = self.state["frame_idx"]
+        frame_group = session.frame_groups[frame_idx]
+
+        for index, instance_group in enumerate(frame_group.instance_groups):
+            if index < 9:
+                shortcut = QKeySequence(Qt.SHIFT + Qt.Key_0 + index + 1)
+            else:
+                shortcut = ""
+            action = QAction(instance_group.name, self.inst_groups_delete_menu, shortcut=shortcut)
+            action.triggered.connect(lambda x, ig=instance_group: self.delete_instance_group(ig))
```

Comment:
You've skipped over using the `CommandContext` as it was intended to be used: add a method `deleteInstanceGroup` to the `CommandContext` and call `self.commands.deleteInstanceGroup` instead of calling this `MainWindow.delete_instance_group`